### PR TITLE
[SPARK-19042] spark executor can't download the jars  when uber jar's http url contains query string

### PR DESCRIPTION
**Pull request by hustfxj**
…ntains any query strings,  the "localName = name.split("/").last" won't get the expected jar, then "val url = new File(SparkFiles.getRootDirectory(), localName).toURI.toURL" will get invalid url. The bug fix is the same as [SPARK-17855]

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Hi, @hustfxj .
If you add `[SPARK-19042]` into your title of this PR as a prefix, the Apache JIRA issue will be changed into `IN PROGRESS'.

**Comment by hustfxj** [comment]
@dongjoon-hyun thank you

**Comment by srowen** [comment]
That seems OK, but where else might a change like this be needed?

**Comment by srowen** [comment]
@hustfxj this might be the only place, but I'd appreciate it if you can search for similar code to see if there are other instances of this issue.

**Comment by hustfxj** [comment]
@hustfxj I searched other places, but not find the similar code.

File: core/src/main/scala/org/apache/spark/executor/Executor.scala [hunk 1]
```diff
@@ -34,12 +34,14 @@ import org.apache.spark.deploy.SparkHadoopUtil
...
 import org.apache.spark.memory.TaskMemoryManager
 import org.apache.spark.rpc.RpcTimeout
-import org.apache.spark.scheduler.{AccumulableInfo, DirectTaskResult, IndirectTaskResult, Task}
+import org.apache.spark.scheduler.{DirectTaskResult, IndirectTaskResult, Task}
 import org.apache.spark.shuffle.FetchFailedException
 import org.apache.spark.storage.{StorageLevel, TaskResultBlockId}
 import org.apache.spark.util._
 import org.apache.spark.util.io.ChunkedByteBuffer
 
+
```
Comment (vanzin) [inline core/src/main/scala/org/apache/spark/executor/Executor.scala]: Don't add these.

**Review by vanzin — COMMENTED** [review]
Can you fix the patch description? It doesn't really follow from the title.

**Comment by hustfxj** [comment]
@vanzin @srowen @dongjoon-hyun can you review  again ? and help  me review [SPARK-18959] and [SPARK-19052], thank you very much.

**Comment by srowen** [comment]
@hustfxj please fix the description

**Comment by hustfxj** [comment]
@srowen I can't  fix the description at this PR, so I close the PR and recreate the new PR for this bug. You can review it at https://github.com/apache/spark/pull/16509.

**Comment by vanzin** [comment]
@hustfxj there should be an edit button for the description just like for every comment (looks like a pencil).

**Pull request closed without merging**